distil/primitives/concat.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+import collections
+import typing
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+
+from d3m import container, utils, exceptions
+from d3m.metadata import base as metadata_base, hyperparams
+from d3m.primitive_interfaces import base, transformer
+from distil.utils import CYTHON_DEP
+
+import version
+
+
+__all__ = ("VerticalConcatenationPrimitive",)
+
+logger = logging.getLogger(__name__)
+
+
+class Hyperparams(hyperparams.Hyperparams):
+    remove_duplicate_rows = hyperparams.Hyperparameter[bool](
+        default=True,
+        semantic_types=[
+            "https://metadata.datadrivendiscovery.org/types/ControlParameter"
+        ],
+        description="If there are two rows with the same d3mIndex, one is retained",
+    )
+    column_overlap = hyperparams.Enumeration[str](
+        default="union",
+        values=("union", "exact", "intersection"),
+        semantic_types=[
+            "https://metadata.datadrivendiscovery.org/types/ControlParameter"
+        ],
+        description="The logic to concat two dataframes.",
+    )
+
+
+class VerticalConcatenationPrimitive(
+    transformer.TransformerPrimitiveBase[
+        container.List, container.DataFrame, Hyperparams
+    ]
+):
+    """
+    A primitive to encapsulate the functionality of pandas.concat.
+    """
+
+    metadata = metadata_base.PrimitiveMetadata(
+        {
+            "id": "b93e3e85-c462-4290-8131-abc51d76a6dd",
+            "version": version.__version__,
+            "name": "DistilRankedLinearSVC",
```

Comment:
incorrect name

Reply:
Whoopsies, guess it's obvious where I copied the template from.